src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackMetricsParser.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using VirtualClient.Common.Contracts;
+    using VirtualClient.Contracts;
+    using static System.Collections.Specialized.BitVector32;
+    using DataTableExtensions = VirtualClient.Contracts.DataTableExtensions;
+
+    /// <summary>
+    /// Parser for Linpack test results file.
+    /// </summary>
+    public class HPLinpackMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// To match metrics line of the result.
+        /// </summary>
+        private const string GetMetricsLine = @"(?<=WR).*(?=\n)";
+
+        /// <summary>
+        /// Split string at one or more spaces.
+        /// </summary>
+        private const string SplitAtSpace = @"\s{1,}";
+
+        /// <summary>
+        /// constructor for <see cref="HPLinpackMetricsParser"/>.
+        /// </summary>
+        /// <param name="rawText">Raw text to parse.</param>
+        public HPLinpackMetricsParser(string rawText)
+            : base(rawText)
+        {
+        }
+
+        /// <summary>
+        /// Results for Linpack results .
+        /// </summary>
+        public DataTable LinpackResult { get; set; }
+
+        private List<Metric> Metrics { get; set; }
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            var matches = Regex.Matches(this.RawText, GetMetricsLine);
+            if (matches.Count == 0)
+            {
+                throw new SchemaException("The HPLinpack output file has incorrect format for parsing");
+            }
+            else
+            {
+                this.Metrics = new List<Metric>();
+                for (int i = 0; i < matches.Count; i++)
+                {
+                    var st = Regex.Split(matches.ElementAt(i).Value, SplitAtSpace);
+
+                    Dictionary<string, IConvertible> metadata = new Dictionary<string, IConvertible>()
+                         {
+                             { "N_WR{st[0]}", st[1] },
+                             { "NB_WR{st[0]}", st[2] },
+                             { "P_WR{st[0]}", st[3] },
+                             { "Q_WR{st[0]}", st[4] },
+                         };
+
+                    this.Metrics.Add(new Metric($"Time", Convert.ToDouble(st[5]), "secs", metadata: metadata));
```

Comment:
Are these N_WR metrics useful? If so, I don't recommend putting them into metadata, they should be proper metrics. If they are not as useful, please at least check them in unit tests.

Reply:
N, NB, P,Q are the input metrics. We can add it as part of scenario, but having them as separate metrics doesn't add much value.

Reply:
ok makes sense then, just verify them in the unit tests, one of each would be enough validation